Dockerfile

```
@@ -1,26 +1,36 @@
-FROM php:7.4
-
-# Install npm
-# libzip-dev: for the PHP zip extension (used by Composer)
-RUN curl -sL https://deb.nodesource.com/setup_12.x | bash - \
-    && apt update && apt install -y git ssh apache2-utils nodejs python3-pip libzip-dev zip && rm -rf /var/lib/apt/lists/*
-
-# Install Composer
-RUN curl -sS https://getcomposer.org/installer | php \
-    && mv composer.phar /usr/local/bin/composer \
-    && chmod +x /usr/local/bin/composer
-
-# Install the `aws` CLI tool
-RUN pip3 install --upgrade --user awscli && echo 'export PATH=/root/.local/bin:$PATH'>/root/.bashrc
-
-# Install serverless
-RUN npm install -g serverless
-
-RUN docker-php-ext-install zip pdo_mysql
-
-RUN mkdir -p /var/task
-
-# Register the Serverless and AWS bin directories
-ENV PATH="/root/.serverless/bin:/root/.local/bin:${PATH}"
-
-WORKDIR '/var/task'
+FROM composer:1.10.4
+
+# composer:1.10.4
+#   https://github.com/composer/docker/blob/master/1.10/Dockerfile
+#   based on php:7-alpine
+#     https://github.com/docker-library/php/blob/master/7.4/alpine3.11/cli/Dockerfile
+#     based on alpine:3.11
+
+# Package Dependencies:
+#   serverless
+#     docker
+#   awscli
+#     groff
+RUN adduser -D -G users -h /home/app app \
+    && composer global require hirak/prestissimo \
+    && apk add --no-cache \
+        docker=19.03.5-r0 \
+        yarn=1.19.2-r0 \
+        groff=1.22.4-r0 \
+    && yarn global add serverless@1.67.0 \
+    && pip3 install \
+        --progress-bar off \
+        --no-cache-dir \
+        --disable-pip-version-check \
+        awscli==1.18.39
+
+USER app
+ENV PATH=$PATH:/tmp/vendor/bin \
+    COMPOSER_ALLOW_SUPERUSER=0 \
+    COMPOSER_HOME=/app/.composer
```

`/app` is where project directory is supposed to be mounted, so that's where `.composer` dir will be created so user can leverage persistent Composer cache in their project.

---

Does that mean that users will get a `.composer` directory in their project? If that's the case, that's something we want to avoid because it will surprise/confuse them.

---

Yes, that's exactly the case. 
I did it on purpose so any PHP package re-installs user runs are as quick as possible. This is with help of Prestissimo and Composer cache which is persistent this way up until user uses the `--rm` flag with `docker run`. So I think having to add one extra line to `.gitignore` is a small price to pay for the benefits. Not to mention that this can be explained in the docs.

```suggestion
    COMPOSER_HOME=/var/task/.composer
```

But if you really think otherwise:
```suggestion
```